src/binary.js

```diff
@@ -120,14 +120,46 @@ function getType(value) {
 }
 
 class Writer {
-    constructor(initialSize) {
-        this.bytes = new Uint8Array(initialSize || 100000000);
-        this.view = new DataView(this.bytes.buffer);
-        this.pos = 0;
+    constructor(chinkSize = 1 * 1024 * 1024 /* 1mb */) {
+        this.chunks = [];
+        this.totalSize = 0;
         this.stringEncoder = new TextEncoder();
+        this.chunkSize = chinkSize;
+        this.createChunk();
     }
     get value() {
-        return this.bytes.subarray(0, this.pos);
+        if (this.pos > 0) {
```

Comment:
This condition looks redundant to me, since there will always be a current chunk here. So just call a `this.flushChunk()` without a condition here